### ISSUE TITLE
process didn't exit successfully: (signal: 9, SIGKILL: kill) 

### ISSUE THREAD

**Issue by laxian** [rust-lang/cargo#13736]
### Problem

I'm new to rust. 
execute `cargo install cargo-ndk` always get error:

```
➜  ~ cargo install cargo-ndk
    Updating crates.io index
  Installing cargo-ndk v3.5.1
    Updating crates.io index
   Compiling proc-macro2 v1.0.79
   Compiling unicode-ident v1.0.12
   Compiling serde v1.0.197
   Compiling syn v1.0.109
   Compiling libc v0.2.153
   Compiling thiserror v1.0.58
   Compiling camino v1.1.6
   Compiling equivalent v1.0.1
   Compiling utf8-decode v1.0.1
   Compiling serde_json v1.0.115
   Compiling semver v1.0.22
   Compiling hashbrown v0.14.3
error: failed to run custom build command for `camino v1.1.6`

Caused by:
  process didn't exit successfully: `/var/folders/03/t8ftyvfj3437x37njdhz9hlm0000gn/T/cargo-installtW38kN/release/build/camino-5633d192f8fd41a1/build-script-build` (signal: 9, SIGKILL: kill)
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `proc-macro2 v1.0.79`

Caused by:
  process didn't exit successfully: `/var/folders/03/t8ftyvfj3437x37njdhz9hlm0000gn/T/cargo-installtW38kN/release/build/proc-macro2-dccb5e150068cb35/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `semver v1.0.22`

Caused by:
  process didn't exit successfully: `/var/folders/03/t8ftyvfj3437x37njdhz9hlm0000gn/T/cargo-installtW38kN/release/build/semver-7ad32938fa4a2502/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `syn v1.0.109`

Caused by:
  process didn't exit successfully: `/var/folders/03/t8ftyvfj3437x37njdhz9hlm0000gn/T/cargo-installtW38kN/release/build/syn-8a34bbb3af64b1db/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to compile `cargo-ndk v3.5.1`, intermediate artifacts can be found at `/var/folders/03/t8ftyvfj3437x37njdhz9hlm0000gn/T/cargo-installtW38kN`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

### Steps

1. Macbook Pro M1 silicon
2. install latest rust: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
3. `cargo install cargo-ndk`

### Version

```text
cargo 1.77.2 (e52e36006 2024-03-26)
release: 1.77.2
commit-hash: e52e360061cacbbeac79f7f1215a7a90b6f08442
commit-date: 2024-03-26
host: aarch64-apple-darwin
libgit2: 1.7.2 (sys:0.18.2 vendored)
libcurl: 8.4.0 (sys:0.4.70+curl-8.5.0 system ssl:(SecureTransport) LibreSSL/3.3.6)
ssl: OpenSSL 1.1.1w  11 Sep 2023
os: Mac OS 14.3.1 [64-bit]
```

**Comment by laxian** [comment]
I got another issue here:
https://github.com/bbqsrc/cargo-ndk/issues/129

**Comment by heisen-li** [comment]
I found ehuss mentioned it before, maybe refer to his [comment](https://github.com/rust-lang/cargo/issues/11641#issuecomment-1407416226) and https://github.com/rust-lang/cargo/issues/8913.

**Comment by weihanglo** [comment]
See https://github.com/rust-lang/cargo/issues/13683#issuecomment-2029846314

tl;dr: check which`strip` you have in your PATH. You should adjust the one from the system, not the one from homebrew which is not compatible.

**Comment by laxian** [comment]
> See [#13683 (comment)](https://github.com/rust-lang/cargo/issues/13683#issuecomment-2029846314)
> 
> tl;dr: check which`strip` you have in your PATH. You should switch to the one from the system, not the one from homebrew which is not compatible.

```
➜  ~ which strip
/opt/homebrew/opt/binutils/bin/strip
➜  ~ whereis strip
strip: /usr/bin/strip /opt/homebrew/opt/binutils/share/man/man1/strip.1
```

**Comment by laxian** [comment]
@weihanglo Thank you, it works.

**Comment by weihanglo** [comment]
Thanks for helping verify it. As it is expected and we've found the root cause, I am going to close this.

**Comment by smartzhao** [comment]
❯ which strip
/opt/homebrew/opt/binutils/bin/strip
❯ brew uninstall binutils
Uninstalling /opt/homebrew/Cellar/binutils/2.43.1... (88 files, 175.6MB)
❯ which strip
/usr/bin/strip

**Comment by GZTimeWalker** [comment]
I'm having the same problem with nix, which is really something to keep an eye on, because it's okay to use `+nightly` compiles...

**Comment by weihanglo** [comment]
https://github.com/NixOS/nixpkgs/issues/299606


It is kinda fixed on nixpkgs with recent versions. I am using nix heavily and can confirm that.
